tlsutil/config.go

```diff
@@ -88,7 +88,7 @@ type ProtocolConfig struct {
 	// certificate authority. This is used to verify authenticity of server
 	// nodes.
 	//
-	// Note: this setting doesn't apply to the gRPC configuration, as Consul
+	// Note: this setting doesn't apply to the external gRPC configuration, as Consul
 	// makes no outgoing connections using this protocol.
```

Comment:
This assertion isn't true anymore, right? Now that Consul is making outgoing gRPC connections for peering replication?

Reply:
We should probably get rid of this check: 

https://github.com/hashicorp/consul/blob/68e79b8180ca89e8cfca291b40a30d943039bd49/agent/config/builder.go#L2522-L2524

And the corresponding tests.

Reply:
This is because the peer conveys dialing instructions via the peering token, rather than us getting any of them from the config file.